tests/unit/test_charm.py

```diff
@@ -232,6 +233,30 @@ def test_honor_labels_is_always_set_in_scrape_configs(self, *unused):
                 self.assertIn("honor_labels", job)
                 self.assertTrue(job["honor_labels"])
 
+    @k8s_resource_multipatch
+    @patch("lightkube.core.client.GenericSyncClient")
+    def test_send_remote_write_endpoints(self, *unused):
+        rel_id = self.harness.add_relation(DEFAULT_CONSUMER_NAME, "prometheus-read")
+        unit_name = "prometheus-read/0"
+        self.harness.add_relation_unit(rel_id, unit_name)
+
+        self.harness.update_relation_data(
+            rel_id,
+            unit_name,
+            {"remote_write": json.dumps({"url": "http://1.1.1.1:9090/api/v1/write"})},
+        )
+
+        container = self.harness.charm.unit.get_container(self.harness.charm._name)
+        config = container.pull(PROMETHEUS_CONFIG)
+        prometheus_scrape_config = yaml.safe_load(config)
+
+        self.assertIn("remote_write", prometheus_scrape_config)
+        self.assertEqual(len(prometheus_scrape_config["remote_write"]), 1)
+        self.assertDictEqual(
+            prometheus_scrape_config["remote_write"][0],
+            {"url": "http://1.1.1.1:9090/api/v1/write"},
+        )
```

Comment:
```suggestion
        self.assertEqual(
            prometheus_scrape_config["remote_write"],
            [{"url": "http://1.1.1.1:9090/api/v1/write"}],
        )
```

Reply:
in this case will not that be an implicit error "Index out of range" if there is no config?